apps/opencs/view/render/terraintexturemode.cpp

```diff
@@ -245,13 +251,12 @@ void CSVRender::TerrainTextureMode::handleDropEvent (QDropEvent *event) {
 void CSVRender::TerrainTextureMode::editTerrainTextureGrid(const WorldspaceHitResult& hit)
 {
     mCellId = getWorldspaceWidget().getCellId (hit.worldPos);
-    std::string hash = "#";
-    std::string space = " ";
-    std::size_t hashlocation = mCellId.find(hash);
-    std::size_t spacelocation = mCellId.find(space);
-    std::string slicedX = mCellId.substr (hashlocation+1, spacelocation-hashlocation);
-    std::string slicedY = mCellId.substr (spacelocation+1);
-    CSMWorld::CellCoordinates mCoordinates(stoi(slicedX), stoi(slicedY));
+
+    std::istringstream streamCoord (mCellId);
+    char ignore;
+    int cellX = 0;
+    int cellY = 0;
+    streamCoord >> ignore >> cellX >> cellY;
```

Comment:
On a second thought, that isn't ideal either: We have the CSMWorld::CellCoordinates class, which makes the whole task much easier (fromId function to create CellCoordinates and then getX/getY). I always forget that we have this class (it was a late addition). Sorry about the confusion.

Reply:
No problem, I'm grateful for learning new stuff along the way.

Reply:
Heh, well, std::pair was a new acquaintance, but here's another solution. The question is - is it better? Could/should that be done without extra variables?

```
std::pair<CSMWorld::CellCoordinates, bool> cellCoordinates_pair = CSMWorld::CellCoordinates::fromId (mCellId);
    CSMWorld::CellCoordinates cellCoordinates = cellCoordinates_pair.first;

    int cellX = cellCoordinates.getX();
    int cellY = cellCoordinates.getY();
```